pkg/segment/query/iqr/intermediateQueryResult.go

```diff
@@ -709,3 +719,47 @@ func (iqr *IQR) AsResult() (*structs.PipeSearchResponseOuter, error) {
 
 	return response, nil
 }
+
+func (iqr *IQR) AddColumnWithDefaultValue(cname string, value utils.CValueEnclosure) error {
```

Comment:
nit: this method looks correct, but probably unnecessary because `AppendKnownValues` does this logic. If you want to keep this method, I'd recommend just doing the `validate()` and then creating the slice, then having `AppendKnownValues` do the rest. That way, there's less code for this IQR implementation, so there's fewer places things can go wrong.

P.S. you can use `toputils.ResizeSliceWithDefault()` on an empty slice to fill it with a default value.
```